src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -170,4 +170,31 @@ public function provideCustomModelTransformerData()
             array('unchecked', false),
         );
     }
+
+    public function testSubmitNullWithEmptyDataTrue()
+    {
+        $form = $this->factory->create('checkbox', null, array(
+            'empty_data' => true,
```

Comment:
But it's valid with default settings, right ?

In that test `$this->assertSame('1', $viewData)` would pass.